azure-spring-boot/src/main/java/com/microsoft/azure/spring/autoconfigure/aad/AADAuthenticationProperties.java

```diff
@@ -31,6 +31,11 @@
     private UserGroupProperties userGroup = new UserGroupProperties();
 
 
+    /**
+     * Default UserGroup configuration for old graph api.
+     */
+    private UserGroupPropertiesOld userGroupOld = new UserGroupPropertiesOld();
```

Comment:
I'm more preferred we can have only one `userGroup` field, instead of using `UserGroupProperties` and `UserGroupPropertiesOld`, how about we have an abstract `AbstractUserGroupProperties`(naming can change), and make an AAD graph user group properties and another microsoft graph user group properties which both extend the `AbstractUserGroupProperties`.

Reply:
so I have gone ahead and changed it to an abstract class with two different property classes that extend this abstract class.  the one issue I was running into was if there are override properties, then I was not able to get these applied.  i tried a multitude of various annotations/etc but couldnt get anything to stick.  the only way i could get this to work was by initializing the usergroup to one of those two property classes at the beginning, and then if setting the usergroup to the other property in the case that microsoft graph api is to be used, then to check to see if any properties were overridden, and if so, to path this into that property class initializer.  all this can be seen in the AADAuthenticationProperties.java file.